File: Modules/SEDO/EMS/SIE.Web.EMS/Checks/Commands/DelMaintainPlanCommand.cs
using SIE.Web.Command;

namespace SIE.Web.EMS.Checks.Commands
{
    /// <summary>
    /// 删除保养明细 命令
    /// </summary>
    [JsCommand(CommandName)]
    public class DelMaintainPlanCommand : ViewCommand
    {
        /// <summary>
        /// 删除保养明细命令名称
        /// </summary>
        public const string CommandName = "SIE.Web.EMS.Checks.Commands.DelMaintainPlanCommand";

        /// <summary>
        /// 执行删除操作
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            return true;
        }
    }
}

File: Modules/SEDO/EMS/SIE.Web.EMS/Checks/Commands/DelMaintainPlanCommand.js
SIE.defineCommand('SIE.Web.EMS.Checks.Commands.DelMaintainPlanCommand', {
    meta: { text: "删除", group: "edit", iconCls: "icon-DeleteEntity icon-red" },
    canExecute: function (view) {
        return view.getSelection().length > 0;
    },
    execute: function (view, source) {
        Ext.MessageBox.confirm("提示".t(), "删除该保养任务同时删除往后的保养任务，是否执行？".t(), function (btnId) {
            if (btnId == "yes") {
                Ext.each(view.getSelection(), function (sel) {
                    var selDateYear = sel.getPlanBeginDate().getYear();
                    var selDateMonth = sel.getPlanBeginDate().getMonth();

                    var entities = [];
                    for (var index = 0; index < view.getData().getData().items.length; index++) {
                        var data = view.getData().getData().items[index];
                        var planBeginDate = data.getPlanBeginDate();
                        if (planBeginDate.getYear() > selDateYear || (planBeginDate.getYear() >= selDateYear && planBeginDate.getMonth() >= selDateMonth)) {
                            entities.push(data);
                        }
                    }
                    for (var index in entities) {
                        view.getData().remove(entities[index]);
                    }
                });
            }
        });
    }
});
